src/Core/Api/Services/FetchDocumentService.php

```diff
@@ -34,39 +36,86 @@ class FetchDocumentService extends APIService implements FetchDocumentContract
    * @throws \Exception
    */
   public function fetch(string $url): ResponseDTO
+  {
+    return $this->fetchInternal($url);
+  }
+
+  /**
+   * Fetch shipping label file from WF server and put it in a ResponseDTO object.
+   * Optionally retry the connection
+   *
+   * @param string $url
+   * @param int $retryRemaining
+   *
+   * @return ResponseDTO
+   * @throws \Exception
+   */
+  private function fetchInternal(string $url, int $retryRemaining = 1): ResponseDTO
   {
     // FIXME: this should be using a generic client via an interface, not cURL!
     $this->loggerContract
       ->debug(TranslationHelper::getLoggerKey('fetchingShipmentForURL'), ['additionalInfo' => ['url' => $url], 'method' => __METHOD__]);
-    $ch = curl_init();
+
     try {
-      if (self::isWayfairAPI($url)) {
-        curl_setopt(
-          $ch, CURLOPT_HTTPHEADER, [
-            'Authorization: ' . $this->authService->generateAuthHeader(),
-            ConfigHelper::WAYFAIR_INTEGRATION_HEADER . ': ' . $this->configHelper->getIntegrationAgentHeader()
-          ]
-        );
+      $ch = curl_init();
```

Comment:
Can we use some http client e.g guzzle instead of curl call?

Reply:
heh you've wandered into this old draft PR, huh?
See line 55: `// FIXME: this should be using a generic client via an interface, not cURL!`